### ISSUE TITLE
Exact Recipe for Training on Coco from scratch

### ISSUE THREAD

**Issue by subhankar-ghosh** [zylo117/Yet-Another-EfficientDet-Pytorch#217]
Can anyone please provide the exact settings (hyperparameters, schedulers, optimizer etc) for achieving the SOTA mAP with any of the architectures (D0, D1 etc any of these). 
With D0 
90k training steps 
Global batch size 16
Number of GPUs 8
================
I am getting loss: 0.315
mAP = 2.716
small mAP = 0.8
med mAP = 2.59
large mAP = 4.45

**Comment by factplay1** [comment]
So I concluded that the code, "as is", does not reproduce COCO training from scratch (efficientdet-0).

Here is what I tried:
- Around 220+ training jobs with different configurations
- hyper-parameter tuning
- 1,2,4,8 GPU cases with various ranges of batch size
- Tried making some changes in the code so it can load pretrained efficientnet-d0 weights. This improved the loss, but still not low enough - lowest it got is 0.305, and even that gives low mAP (4%)
- Tried different kinds of optimizers, different kinds of LR scheduling techniques
- Tried 2-step training, where you train with adamw first, then try to further finetune with sgd
- Tried with the latest code as well as throughout the past 3 weeks or so
- Tried random seed vs fixed seed
- Tried different kinds of pretrained weights

I concluded that for custom datasets it works, but not for COCO training from scratch for research purposes.

**Comment by claire-s11** [comment]
Hi @subhankar-ghosh , by mAP = 2.716, do you mean iou= 0.5:0.95, map = 0.2716? Thank you!
